Fix tests that sometimes fail

Some tests usually succeed, but sometimes don't (e.g. under high cpu load, some timeouts are set too low).

Examples:

(this first one happens a lot)
```
- must return valid log messages for each status event *** FAILED *** (10 seconds, 354 milliseconds)
  java.lang.AssertionError: Timeout (10 seconds) waiting for 1 messages on LoggingTestKitImpl(1,Some(INFO),None,None,Some(Simulating from 2011-01-01 00:00:00 until 2011-01-01 01:00:00.),None,None,Map(),true,None).
  at akka.actor.testkit.typed.internal.LoggingTestKitImpl.expect(LoggingTestKitImpl.scala:103)
  at edu.ie3.simona.event.listener.RuntimeEventListenerSpec.$anonfun$new$5(RuntimeEventListenerSpec.scala:155)
  at edu.ie3.simona.event.listener.RuntimeEventListenerSpec.$anonfun$new$5$adapted(RuntimeEventListenerSpec.scala:151)
  at scala.collection.immutable.List.foreach(List.scala:333)
  at edu.ie3.simona.event.listener.RuntimeEventListenerSpec.$anonfun$new$4(RuntimeEventListenerSpec.scala:151)
  at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
  at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
  at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
  at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
  at org.scalatest.Transformer.apply(Transformer.scala:22)
  ...
```

```
- should start the simulation, do 2 sweeps and should end afterwards when no deviation on nodal power is recognized in the superior when a StartGridSimulationTrigger is send *** FAILED *** (4 seconds, 22 milliseconds)
  java.lang.AssertionError: assertion failed: timeout (3 seconds) during expectMsg:
  at scala.Predef$.assert(Predef.scala:279)
  at akka.testkit.TestKitBase.expectMsgPF(TestKit.scala:479)
  at akka.testkit.TestKitBase.expectMsgPF$(TestKit.scala:476)
  at akka.testkit.TestKit.expectMsgPF(TestKit.scala:973)
  at edu.ie3.simona.agent.grid.DBFSAlgorithmSupGridSpec.$anonfun$new$6(DBFSAlgorithmSupGridSpec.scala:189)
  at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:190)
  at edu.ie3.simona.agent.grid.DBFSAlgorithmSupGridSpec.$anonfun$new$5(DBFSAlgorithmSupGridSpec.scala:148)
  at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
  at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
  at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
```

```
- should write a series of new NodeResults to Kafka *** FAILED *** (22 seconds, 213 milliseconds)
  The code passed to eventually never returned normally. Attempted 13 times over 21.375327958000003 seconds. Last failure message: List() had length 0 instead of expected length 3. (ResultEntityKafkaSpec.scala:156)
  org.scalatest.exceptions.TestFailedDueToTimeoutException:
  ...
  at edu.ie3.simona.io.result.ResultEntityKafkaSpec.$anonfun$new$2(ResultEntityKafkaSpec.scala:156)
  at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
  at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
  at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
  at org.scalatest.Transformer.apply(Transformer.scala:22)
  at org.scalatest.Transformer.apply(Transformer.scala:20)
  at org.scalatest.wordspec.AnyWordSpecLike$$anon$3.apply(AnyWordSpecLike.scala:1239)
  ...
  Cause: org.scalatest.exceptions.TestFailedException: List() had length 0 instead of expected length 3
  ...
  at edu.ie3.simona.io.result.ResultEntityKafkaSpec.$anonfun$new$3(ResultEntityKafkaSpec.scala:160)
  at org.scalatest.enablers.Retrying$$anon$4.makeAValiantAttempt$1(Retrying.scala:184)
  at org.scalatest.enablers.Retrying$$anon$4.tryTryAgain$2(Retrying.scala:196)
  at org.scalatest.enablers.Retrying$$anon$4.retry(Retrying.scala:226)
  at org.scalatest.concurrent.Eventually.eventually(Eventually.scala:313)
  at org.scalatest.concurrent.Eventually.eventually$(Eventually.scala:312)
  at edu.ie3.simona.io.result.ResultEntityKafkaSpec.eventually(ResultEntityKafkaSpec.scala:43)
```

```
  - should shutdown and compress the data when requested to do so without any errors *** FAILED *** (10 seconds, 386 milliseconds)
    java.lang.AssertionError: assertion failed: timeout 10 seconds expired:
    at scala.Predef$.assert(Predef.scala:279)
    at akka.testkit.TestKitBase.poll$1(TestKit.scala:300)
    at akka.testkit.TestKitBase.awaitCond(TestKit.scala:306)
    at akka.testkit.TestKitBase.awaitCond$(TestKit.scala:289)
    at akka.testkit.TestKit.awaitCond(TestKit.scala:973)
    at edu.ie3.simona.event.listener.ResultEventListenerSpec.$anonfun$new$37(ResultEventListenerSpec.scala:563)
    at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
    at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
    at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
    at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
    ...
```

I found another one.
    
    - should process a valid participants result correctly *** FAILED *** (10 seconds, 23 milliseconds)
          java.lang.AssertionError: assertion failed: timeout 10 seconds expired:
          at scala.Predef$.assert(Predef.scala:279)
          at akka.testkit.TestKitBase.poll$1(TestKit.scala:300)
          at akka.testkit.TestKitBase.awaitCond(TestKit.scala:306)
          at akka.testkit.TestKitBase.awaitCond$(TestKit.scala:289)
          at akka.testkit.TestKit.awaitCond(TestKit.scala:973)
          at edu.ie3.simona.event.listener.ResultEventListenerSpec.$anonfun$new$7(ResultEventListenerSpec.scala:192)  
          at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
          at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
          at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
          at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
          ...

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -88,6 +88,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Fixed groovy formatting [#110](https://github.com/ie3-institute/simona/issues/110)
 - Fixed configuration reference in user's guide [#224](https://github.com/ie3-institute/simona/issues/224)
 - Fixed ResultEventListener exiting too early with high volumes of results [#350](https://github.com/ie3-institute/simona/issues/350)
+- Fixed tests that unreliably fail [#359](https://github.com/ie3-institute/simona/issues/359)
 
 ### Removed
 - Remove workaround for tscfg tmp directory [#178](https://github.com/ie3-institute/simona/issues/178)

--- a/src/main/scala/edu/ie3/simona/event/listener/ResultEventListener.scala
+++ b/src/main/scala/edu/ie3/simona/event/listener/ResultEventListener.scala
@@ -8,7 +8,6 @@ package edu.ie3.simona.event.listener
 
 import akka.actor._
 import akka.pattern.pipe
-import akka.stream.Materializer
 import edu.ie3.datamodel.io.processor.result.ResultEntityProcessor
 import edu.ie3.datamodel.models.result.{NodeResult, ResultEntity}
 import edu.ie3.simona.agent.grid.GridResultsSupport.PartialTransformer3wResult
@@ -94,8 +93,6 @@ object ResultEventListener extends Transformer3wResultSupport {
     */
   private def initializeSinks(
       resultFileHierarchy: ResultFileHierarchy
-  )(implicit
-      materializer: Materializer
   ): Iterable[Future[(Class[_], ResultEntitySink)]] = {
     resultFileHierarchy.resultSinkType match {
       case _: ResultSinkType.Csv =>
@@ -176,8 +173,6 @@ class ResultEventListener(
     with FSM[AgentState, ResultEventListenerData]
     with SimonaFSMActorLogging
     with Stash {
-
-  implicit private val materializer: Materializer = Materializer(context)
 
   override def preStart(): Unit = {
     log.debug("Starting initialization!")
@@ -321,6 +316,11 @@ class ResultEventListener(
       stash()
       stay()
 
+    case Event(StopMessage(_), _) =>
+      // set ReceiveTimeout message to be sent if no message has been received for 5 seconds
+      stash()
+      stay()
+
     case Event(Init, _) =>
       Future
         .sequence(
@@ -341,6 +341,7 @@ class ResultEventListener(
 
     case Event(Status.Failure(ex), _) =>
       throw new InitializationException("Unable to setup SimonaSim.", ex)
+
   }
 
   when(Idle) {

--- a/src/main/scala/edu/ie3/simona/event/listener/ResultEventListener.scala
+++ b/src/main/scala/edu/ie3/simona/event/listener/ResultEventListener.scala
@@ -317,7 +317,6 @@ class ResultEventListener(
       stay()
 
     case Event(StopMessage(_), _) =>
-      // set ReceiveTimeout message to be sent if no message has been received for 5 seconds
       stash()
       stay()
 

--- a/src/test/scala/edu/ie3/simona/agent/grid/DBFSAlgorithmFailedPowerFlowSpec.scala
+++ b/src/test/scala/edu/ie3/simona/agent/grid/DBFSAlgorithmFailedPowerFlowSpec.scala
@@ -35,6 +35,7 @@ import edu.ie3.simona.test.common.model.grid.DbfsTestGrid
 import edu.ie3.simona.test.common.{ConfigTestData, TestKitWithShutdown}
 import edu.ie3.util.quantities.QuantityUtils.RichQuantityDouble
 
+import scala.concurrent.duration.DurationInt
 import scala.language.postfixOps
 
 class DBFSAlgorithmFailedPowerFlowSpec
@@ -238,7 +239,8 @@ class DBFSAlgorithmFailedPowerFlowSpec
 
       // the requested power is to high for the grid to handle, therefore the superior grid agent
       // receives a FailedPowerFlow message
-      superiorGridAgent.gaProbe.expectMsg(FailedPowerFlow)
+      // wait 30 seconds max for power flow to finish
+      superiorGridAgent.gaProbe.expectMsg(30 seconds, FailedPowerFlow)
 
       // normally the slack node would send a FinishGridSimulationTrigger to all
       // connected inferior grids, because the slack node is just a mock, we imitate this behavior

--- a/src/test/scala/edu/ie3/simona/agent/grid/DBFSAlgorithmSupGridSpec.scala
+++ b/src/test/scala/edu/ie3/simona/agent/grid/DBFSAlgorithmSupGridSpec.scala
@@ -42,6 +42,7 @@ import tech.units.indriya.quantity.Quantities
 
 import java.util.UUID
 import scala.concurrent.duration.DurationInt
+import scala.language.postfixOps
 
 /** Test to ensure the functions that a [[GridAgent]] in superior position
   * should be able to do if the DBFSAlgorithm is used. The scheduler, the
@@ -206,7 +207,8 @@ class DBFSAlgorithmSupGridSpec
 
           // we expect a completion message here and that the agent goes back to simulate grid
           // and waits until the newly scheduled StartGridSimulationTrigger is send
-          scheduler.expectMsgPF() {
+          // wait 30 seconds max for power flow to finish
+          scheduler.expectMsgPF(30 seconds) {
             case CompletionMessage(
                   2,
                   Some(
@@ -362,7 +364,8 @@ class DBFSAlgorithmSupGridSpec
           // and waits until the newly scheduled StartGridSimulationTrigger is send
 
           // Simulate Grid
-          scheduler.expectMsgPF(30.seconds) {
+          // wait 30 seconds max for power flow to finish
+          scheduler.expectMsgPF(30 seconds) {
             case CompletionMessage(
                   _,
                   Some(

--- a/src/test/scala/edu/ie3/simona/agent/grid/DBFSMockGridAgents.scala
+++ b/src/test/scala/edu/ie3/simona/agent/grid/DBFSMockGridAgents.scala
@@ -21,7 +21,8 @@ import edu.ie3.simona.ontology.messages.VoltageMessage.{
 import edu.ie3.simona.test.common.UnitSpec
 
 import java.util.UUID
-import scala.concurrent.duration.DurationInt
+import scala.concurrent.duration.{DurationInt, FiniteDuration}
+import scala.language.postfixOps
 
 /** Provide mock grid agents for testing the DBFSAlgorithm. These agents are an
   * agent for inferior grids and an agent for superior grids. Each grid agent
@@ -109,9 +110,10 @@ trait DBFSMockGridAgents extends UnitSpec {
     }
 
     def expectGridPowerProvision(
-        expectedExchangedPowers: Seq[ExchangePower]
+        expectedExchangedPowers: Seq[ExchangePower],
+        maxDuration: FiniteDuration = 30 seconds
     ): Unit = {
-      inside(gaProbe.expectMsgType[ProvideGridPowerMessage](10.seconds)) {
+      inside(gaProbe.expectMsgType[ProvideGridPowerMessage](maxDuration)) {
         case ProvideGridPowerMessage(exchangedPower) =>
           exchangedPower should have size expectedExchangedPowers.size
 

--- a/src/test/scala/edu/ie3/simona/agent/participant/ParticipantAgentExternalSourceSpec.scala
+++ b/src/test/scala/edu/ie3/simona/agent/participant/ParticipantAgentExternalSourceSpec.scala
@@ -227,6 +227,8 @@ class ParticipantAgentExternalSourceSpec
                       actorToBeScheduled
                     ) =>
                   tick == 4711L && actorToBeScheduled == mockAgent
+                case unexpected =>
+                  fail(s"Received unexpected trigger message $unexpected")
               } shouldBe true
             case None => fail("Expected to get a trigger for tick 4711.")
           }

--- a/src/test/scala/edu/ie3/simona/event/listener/ResultEventListenerSpec.scala
+++ b/src/test/scala/edu/ie3/simona/event/listener/ResultEventListenerSpec.scala
@@ -71,7 +71,7 @@ class ResultEventListenerSpec
     classOf[LineResult]
   )
 
-  private val timeout = 60 seconds
+  private val timeout = 10 minutes
 
   // the OutputFileHierarchy
   private def resultFileHierarchy(

--- a/src/test/scala/edu/ie3/simona/event/listener/ResultEventListenerSpec.scala
+++ b/src/test/scala/edu/ie3/simona/event/listener/ResultEventListenerSpec.scala
@@ -71,7 +71,7 @@ class ResultEventListenerSpec
     classOf[LineResult]
   )
 
-  private val timeout = 20 seconds
+  private val timeout = 60 seconds
 
   // the OutputFileHierarchy
   private def resultFileHierarchy(

--- a/src/test/scala/edu/ie3/simona/event/listener/ResultEventListenerSpec.scala
+++ b/src/test/scala/edu/ie3/simona/event/listener/ResultEventListenerSpec.scala
@@ -71,7 +71,7 @@ class ResultEventListenerSpec
     classOf[LineResult]
   )
 
-  private val timeout = 10.seconds
+  private val timeout = 20 seconds
 
   // the OutputFileHierarchy
   private def resultFileHierarchy(
@@ -472,7 +472,8 @@ class ResultEventListenerSpec
         /* The result file is created at start up and only contains a head line. */
         awaitCond(
           outputFile.exists(),
-          interval = 500.millis
+          interval = 500.millis,
+          max = timeout
         )
         getFileLinesLength(outputFile) shouldBe 1
 
@@ -513,7 +514,8 @@ class ResultEventListenerSpec
         /* Await that the result is written */
         awaitCond(
           getFileLinesLength(outputFile) == 2,
-          interval = 500.millis
+          interval = 500.millis,
+          max = timeout
         )
         /* Check the result */
         val resultFileSource = Source.fromFile(outputFile)
@@ -561,7 +563,7 @@ class ResultEventListenerSpec
           )
         )
 
-        awaitCond(outputFile.exists(), interval = 500.millis)
+        awaitCond(outputFile.exists(), interval = 500.millis, max = timeout)
 
         // graceful shutdown should wait until existing messages within an actor are fully processed
         // otherwise it might happen, that the shutdown is triggered even before the just send ParticipantResultEvent

--- a/src/test/scala/edu/ie3/simona/event/listener/ResultEventListenerSpec.scala
+++ b/src/test/scala/edu/ie3/simona/event/listener/ResultEventListenerSpec.scala
@@ -7,8 +7,6 @@
 package edu.ie3.simona.event.listener
 
 import akka.actor.ActorSystem
-import akka.pattern.gracefulStop
-import akka.stream.Materializer
 import akka.testkit.{TestFSMRef, TestProbe}
 import com.typesafe.config.ConfigFactory
 import edu.ie3.datamodel.models.result.connector.{
@@ -25,6 +23,7 @@ import edu.ie3.simona.event.ResultEvent.{
   PowerFlowResultEvent
 }
 import edu.ie3.simona.io.result.{ResultEntitySink, ResultSinkType}
+import edu.ie3.simona.ontology.messages.StopMessage
 import edu.ie3.simona.test.common.result.PowerFlowResultData
 import edu.ie3.simona.test.common.{AgentSpec, IOTestCommons, UnitSpec}
 import edu.ie3.simona.util.ResultFileHierarchy
@@ -71,7 +70,7 @@ class ResultEventListenerSpec
     classOf[LineResult]
   )
 
-  private val timeout = 60 seconds
+  private val timeout = 20 seconds
 
   // the OutputFileHierarchy
   private def resultFileHierarchy(
@@ -92,17 +91,12 @@ class ResultEventListenerSpec
   def createDir(
       resultFileHierarchy: ResultFileHierarchy
   ): Iterable[Future[ResultEntitySink]] = {
-    val materializer: Materializer = Materializer(system)
-
     val initializeSinks: PrivateMethod[Iterable[Future[ResultEntitySink]]] =
       PrivateMethod[Iterable[Future[ResultEntitySink]]](
         Symbol("initializeSinks")
       )
 
-    ResultEventListener invokePrivate initializeSinks(
-      resultFileHierarchy,
-      materializer
-    )
+    ResultEventListener invokePrivate initializeSinks(resultFileHierarchy)
   }
 
   private def getFileLinesLength(file: File) = {
@@ -183,10 +177,7 @@ class ResultEventListenerSpec
         awaitCond(outputFile.exists(), interval = 500.millis, max = timeout)
 
         // stop listener so that result is flushed out
-        Await.ready(
-          gracefulStop(listenerRef, timeout),
-          timeout
-        )
+        listenerRef ! StopMessage(true)
 
         // wait until all lines have been written out:
         awaitCond(
@@ -269,10 +260,7 @@ class ResultEventListenerSpec
         )
 
         // stop listener so that result is flushed out
-        Await.ready(
-          gracefulStop(listenerRef, timeout),
-          timeout
-        )
+        listenerRef ! StopMessage(true)
 
         // wait until all lines have been written out:
         awaitCond(
@@ -506,10 +494,7 @@ class ResultEventListenerSpec
         )
 
         // stop listener so that result is flushed out
-        Await.ready(
-          gracefulStop(listener, timeout),
-          timeout
-        )
+        listener ! StopMessage(true)
 
         /* Await that the result is written */
         awaitCond(
@@ -565,18 +550,11 @@ class ResultEventListenerSpec
 
         awaitCond(outputFile.exists(), interval = 500.millis, max = timeout)
 
-        // graceful shutdown should wait until existing messages within an actor are fully processed
+        // stopping the actor should wait until existing messages within an actor are fully processed
         // otherwise it might happen, that the shutdown is triggered even before the just send ParticipantResultEvent
         // reached the listener
         // this also triggers the compression of result files
-        import akka.pattern._
-        Await.ready(
-          gracefulStop(listenerRef, timeout),
-          timeout
-        )
-
-        // shutdown the actor system
-        Await.ready(system.terminate(), 1.minute)
+        listenerRef ! StopMessage(true)
 
         // wait until file exists
         awaitCond(

--- a/src/test/scala/edu/ie3/simona/event/listener/ResultEventListenerSpec.scala
+++ b/src/test/scala/edu/ie3/simona/event/listener/ResultEventListenerSpec.scala
@@ -71,7 +71,7 @@ class ResultEventListenerSpec
     classOf[LineResult]
   )
 
-  private val timeout = 10 minutes
+  private val timeout = 60 seconds
 
   // the OutputFileHierarchy
   private def resultFileHierarchy(

--- a/src/test/scala/edu/ie3/simona/event/listener/RuntimeEventListenerSpec.scala
+++ b/src/test/scala/edu/ie3/simona/event/listener/RuntimeEventListenerSpec.scala
@@ -35,8 +35,10 @@ import java.util.concurrent.{LinkedBlockingQueue, TimeUnit}
 class RuntimeEventListenerSpec
     extends ScalaTestWithActorTestKit(
       ActorTestKit.ApplicationTestConfig.withValue(
+        // Timeout for LoggingTestKit via TestKitSettings
+        // Log message sometimes seem to take a while until caught by the test kit
         "akka.actor.testkit.typed.filter-leeway",
-        ConfigValueFactory.fromAnyRef("10s")
+        ConfigValueFactory.fromAnyRef("30s")
       )
     )
     with AnyWordSpecLike

--- a/src/test/scala/edu/ie3/simona/io/result/ResultEntityKafkaSpec.scala
+++ b/src/test/scala/edu/ie3/simona/io/result/ResultEntityKafkaSpec.scala
@@ -153,7 +153,8 @@ class ResultEntityKafkaSpec
       Then("records can be fetched from Kafka")
       testConsumer.seekToBeginning(topicPartitions.asJava)
 
-      eventually(timeout(20 seconds), interval(1 second)) {
+      // kafka messages might take some time if machine is loaded
+      eventually(timeout(60 seconds), interval(1 second)) {
         val records: List[PlainNodeResult] =
           testConsumer.poll((1 second) toJava).asScala.map(_.value()).toList
 

--- a/src/test/scala/edu/ie3/simona/io/result/ResultEntityKafkaSpec.scala
+++ b/src/test/scala/edu/ie3/simona/io/result/ResultEntityKafkaSpec.scala
@@ -154,7 +154,7 @@ class ResultEntityKafkaSpec
       testConsumer.seekToBeginning(topicPartitions.asJava)
 
       // kafka messages might take some time if machine is loaded
-      eventually(timeout(60 seconds), interval(1 second)) {
+      eventually(timeout(2 minutes), interval(1 second)) {
         val records: List[PlainNodeResult] =
           testConsumer.poll((1 second) toJava).asScala.map(_.value()).toList
 

--- a/src/test/scala/edu/ie3/simona/test/KafkaSpecLike.scala
+++ b/src/test/scala/edu/ie3/simona/test/KafkaSpecLike.scala
@@ -9,10 +9,10 @@ package edu.ie3.simona.test
 import com.dimafeng.testcontainers.KafkaContainer
 import edu.ie3.simona.test.KafkaSpecLike.Topic
 import org.apache.kafka.clients.admin.{Admin, NewTopic}
-import org.junit.Rule
 import org.scalatest.{BeforeAndAfterAll, TestSuite}
 import org.testcontainers.utility.DockerImageName
 
+import java.util.concurrent.TimeUnit
 import scala.jdk.CollectionConverters._
 
 /** Adapted from
@@ -23,7 +23,6 @@ trait KafkaSpecLike extends BeforeAndAfterAll {
 
   protected val testTopics: Seq[Topic]
 
-  @Rule
   protected val kafka: KafkaContainer = KafkaContainer(
     DockerImageName.parse("confluentinc/cp-kafka:6.1.0")
   )
@@ -34,7 +33,7 @@ trait KafkaSpecLike extends BeforeAndAfterAll {
   override def beforeAll(): Unit = {
     super.beforeAll()
     kafka.start()
-    admin.createTopics(
+    val result = admin.createTopics(
       testTopics.map { topic =>
         new NewTopic(
           topic.name,
@@ -43,6 +42,9 @@ trait KafkaSpecLike extends BeforeAndAfterAll {
         )
       }.asJava
     )
+
+    // wait for result, throw exception if applicable
+    result.all().get(1, TimeUnit.MINUTES)
   }
 
   override def afterAll(): Unit = {

--- a/src/test/scala/edu/ie3/simona/test/KafkaSpecLike.scala
+++ b/src/test/scala/edu/ie3/simona/test/KafkaSpecLike.scala
@@ -24,7 +24,7 @@ trait KafkaSpecLike extends BeforeAndAfterAll {
   protected val testTopics: Seq[Topic]
 
   protected val kafka: KafkaContainer = KafkaContainer(
-    DockerImageName.parse("confluentinc/cp-kafka:6.1.0")
+    DockerImageName.parse("confluentinc/cp-kafka:7.3.1")
   )
   protected lazy val admin: Admin = Admin.create(
     Map[String, AnyRef]("bootstrap.servers" -> kafka.bootstrapServers).asJava

